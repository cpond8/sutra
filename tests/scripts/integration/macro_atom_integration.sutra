;; Integration test - macros and atoms working together
;; Tests integration between macro system and atom operations

;; Macro that uses arithmetic atoms
(define (double x) (* x 2))
(print (double 5))
(print (double 10))

;; Macro that uses comparison atoms
(define (is_positive x) (gt? x 0))
(print (is_positive 5))
(print (is_positive -3))

;; Macro that uses path operations
(define (store_and_get key value)
  (do
    (set! key value)
    (get key)))
(print (store_and_get test.item "stored_value"))

;; Macro that combines multiple operations
(define (calculate_and_store x y)
  (do
    (set! result (+ x y))
    (print (get result))))
(calculate_and_store 10 20)
