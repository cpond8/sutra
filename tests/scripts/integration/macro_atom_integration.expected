10
20
true
false
stored_value
30
